src/SBarcodeDecoder.cpp

```diff
@@ -190,52 +200,37 @@ QImage SBarcodeDecoder::videoFrameToImage(QVideoFrame &videoFrame, const QRect &
         return image.rgbSwapped().copy(captureRect);
     }
 
-    return QImage();
-} // SBarcodeDecoder::videoFrameToImage
+    #else
 
-QImage SBarcodeDecoder::imageFromVideoFrame(const QVideoFrame &videoFrame)
-{
-    uchar *ARGB32Bits = new uchar[(videoFrame.width() * videoFrame.height()) * 4];
-
-    QImage::Format imageFormat = videoFrame.imageFormatFromPixelFormat(videoFrame.pixelFormat());
-
-    if (imageFormat == QImage::Format_Invalid) {
-        switch (videoFrame.pixelFormat()) {
-            case QVideoFrame::Format_YUYV: qt_convert_YUYV_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_NV12: qt_convert_NV12_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_YUV420P: qt_convert_YUV420P_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_YV12: qt_convert_YV12_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_AYUV444: qt_convert_AYUV444_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_YUV444: qt_convert_YUV444_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_UYVY: qt_convert_UYVY_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_NV21: qt_convert_NV21_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_BGRA32: qt_convert_BGRA32_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_BGR24: qt_convert_BGR24_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_BGR565: qt_convert_BGR565_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            case QVideoFrame::Format_BGR555: qt_convert_BGR555_to_ARGB32(videoFrame, ARGB32Bits);
-                break;
-            default: break;
+    auto handleType = videoFrame.handleType();
+
+    if (handleType == QVideoFrame::NoHandle) {
+
+        QImage image = videoFrame.toImage();
+
+        if (image.isNull()) {
+            return QImage();
+        }
+
+        if (image.format() != QImage::Format_ARGB32) {
+            image = image.convertToFormat(QImage::Format_ARGB32);
         }
 
-        return QImage(ARGB32Bits,
-                 videoFrame.width(),
-                 videoFrame.height(),
-                 QImage::Format_ARGB32);
+        // that's because qml videooutput has no mapNormalizedRectToItem method
+#ifdef Q_OS_ANDROID
+        return image.copy(m_resolutionHeight/4, m_resolutionWidth/4, m_resolutionHeight/2, m_resolutionWidth/2);
+#else
+        return image.copy(captureRect);
+#endif
```

Comment:
It is very bad solution. You should be used the [cintentRect](https://doc.qt.io/qt-6/qml-qtmultimedia-videooutput.html#contentRect-prop) property from the qml video output and map this to scaner rectangel.

Reply:
yeah I would like to, but Qt6 VideoOutput element has no mapping methods